### PR TITLE
docs: document kubelet cgroup driver detection from the runtime

### PR DESCRIPTION
**Pull request by marquiz**
Document the KubeletCgroupDriverFromCRI feature gate. Also, add notes of this feature in parts of the documentation that describe cgroup driver configuration.

Refs https://github.com/kubernetes/enhancements/issues/4033

NOTE: no container runtime supports this new feature, yet, so we cannot refer to specific versions of runtimes (yet)

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -561,6 +562,13 @@ Each feature gate is designed for enabling/disabling a specific feature:
   The Job controller uses Pod finalizers and a field in the Job status to keep
   track of the finished Pods to count towards completion.
 - `KMSv2`: Enables KMS v2 API for encryption at rest. See [Using a KMS Provider for data encryption](/docs/tasks/administer-cluster/kms-provider) for more details.
+- `KubeletCgroupDriverFromCRI`: Enable detection of the kubelet cgroup driver
+  configuration option from the CRI. The container runtime also needs to support
```
Comment (haircommander) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:566]: I find the sentence 
```
The container runtime also needs to support this feature in which case the kubelet will ignore the cgroupDriver...
```

to be kind of clunky. Maybe 
```
This feature gate requires the user run a container runtime that supports the `RuntimeConfig` CRI call. If both CRI and Kubelet support this feature, the kubelet will ignore the cgroupDriver (--cgroup-driver) configuration optoin. ...

```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -561,6 +562,13 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  this feature in which case the kubelet will ignore the cgroupDriver
+  (--cgroup-driver) configuration option. If the container runtime doesn't
+  support it, the kubelet will fallback to using it's cgroupDriver option.
```
Comment (windsonsea) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:569]:
```suggestion
  support it, the kubelet will fallback to using its cgroupDriver option.
```

Comment (marquiz) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Thanks @windsonsea, fixed

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (marquiz) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: @haircommander changed as suggested. I left the last sentence (`If the container runtime doesn't support it, the kubelet will fallback to using its cgroupDriver option.`) intact.

**Comment by haircommander** [comment]
LGTM, thanks!

**Comment by marquiz** [comment]
Woops, submitted against the wrong branch 🙄 Replaced by #42160
/close

**Pull request closed without merging**